### ISSUE TITLE
Training a custom dataset for deep sort

### ISSUE THREAD

**Issue by tonmoyborah** [nwojke/cosine_metric_learning#5]
I want to use deep sort for tracking objects other than people (say cars). I presume I have to train this cosine metric model to generate a .pb file to use in deep sort? Now to train this model, what is the format or structure of dataset/folders that I need? And what type of data do I need? Is a set of images with their gt_boxes enough?
What I can understand is, I need to have a directory of frames from videos ( training data images) and the gt_boxes 
-->  create `.pb` file by training cosine metric model on train data (on cropped images of specific objects(?)) 
--> train Faster RCNN model using training data 
--> generate predictions using Faster RCNN on test data 
--> create `.npy` files on test image directory and predicted boxes with `generate_detections.py` from deep sort 
--> feed them into `deep_sort_app.py`.
 Anything else I need to take care of?
Thanks for the great work!

**Comment by nwojke** [comment]
I think you got it quite right. The training code expects (cropped) images (of specific objects) and associated class labels, i.e., one label for each identity in the dataset.

**Comment by nwojke** [comment]
Also, you need to write a Python script to read the dataset and call the trainer. I have added code for training on [VeRi](https://github.com/VehicleReId/VeRidataset). Check out this [commit](https://github.com/nwojke/cosine_metric_learning/commit/f44d01d4fcdf42bdcaa4b6bfee5762d56a145258).

**Comment by tonmoyborah** [comment]
Thanks a lot! 
Also, I have another doubt.. I don't quite understand the purpose of the evaluation software of MARS dataset and is it necessary? (I want to be sure before downloading the huge dataset)

**Comment by nwojke** [comment]
You only need the MARS dataset if you plan to re-train on this particular data. You only need the MARS evaluation software if you plan to evaluate a trained model on the MARS test data.

**Comment by tshan1229** [comment]
I know I'm late but I just found out about this post and I am looking to train my model for detecting logos.
I've successfully trained my model for Yolov3 and I've images with labels using [this](https://github.com/AlexeyAB/Yolo_mark). 
1> Will this dataset work for cosine metric model? 
2> After that how can I train using Faster CNN and use .pb file previously generated.
3> Once done with training I assume it'll also have steps on how to generate predictions on the training data.
4> @nwojke you wrote this Also, you need to write a Python script to read the dataset and call the trainer. I have added code for training on VeRi. Check out this commit.
When do I need to this step before step 1?

Any help will be highly appreciated.

Thaks 
Shan

**Comment by SamrudaPobbathi** [comment]
What kind of input format does the script expect? I am trying to use the code, and am finding issues with choosing batch sizes and num_samples_per_id. 

> Also, you need to write a Python script to read the dataset and call the trainer. I have added code for training on [VeRi](https://github.com/VehicleReId/VeRidataset). Check out this [commit](https://github.com/nwojke/cosine_metric_learning/commit/f44d01d4fcdf42bdcaa4b6bfee5762d56a145258).

I am confused about the components needed in trainer script. Can you kindly elaborate on the structure of the dataset folder needed and the train_veri more? Thanks in advance

**Comment by jhonjam** [comment]
hi i am using yolov3 for vehicle detection and deep sort for tracking. however in some frames the detection of an object is lost, the deep sort algorithm generates a new id for the same object. I would like to do a new cosine metric model training to generate a .pb file to use in deep sort with the data set VeRI [](https://github.com/JDAI-CV/VeRidataset), however I have no idea what the format of the ground truth of objects is, in yolo the format is class, x1, y1, x2, y2, to train "cosine metric model" how would the gt_boxes of the images be? Could you explain or give me an idea about this. or someone who can share the trained model with me.

Thank you

**Comment by priyanka36** [comment]
I want to do post training quantization on the deepsort model.I need the representative dataset for the process . Can I get few dataset for the purpose in ranges of 200-500 images ? If I can then where will I find that?
